tests/brain/synapses.cpp

```diff
@@ -0,0 +1,221 @@
+
+        for( size_t j = 0; j < synapses.size(); ++j )
+            bbox.merge( vmml::Vector3f( posx[j], posy[j], posz[j] ));
+        totalSize += synapses.size();
+    }
+
+    BOOST_CHECK_CLOSE( bbox.getCenter()[0], 19.4931183f, 0.00001f );
+    BOOST_CHECK_CLOSE( bbox.getCenter()[1], 1384.17578f, 0.00001f );
+    BOOST_CHECK_CLOSE( bbox.getCenter()[2], 18.0030212f, 0.00001f );
+    BOOST_CHECK_EQUAL( totalSize, 9520 );
+}
+
+BOOST_AUTO_TEST_CASE( afferent_synapses )
+{
+    const brain::Circuit circuit( brion::URI( BBP_TEST_BLUECONFIG3 ));
+    const brain::Synapses& synapses =
+            circuit.getAfferentSynapses( circuit.getGIDs( "Layer1" ));
+    BOOST_CHECK( !synapses.empty( ));
+    BOOST_CHECK_EQUAL( synapses.size(), 1172 );
+    BOOST_CHECK_EQUAL( synapses[0].getPresynapticGID(), 10 );
+    BOOST_CHECK_CLOSE( synapses[1].getPostsynapticDistance(), 1.34995711f,
+                       0.00001f );
+    BOOST_CHECK_CLOSE( synapses[2].getConductance(), 0.34758395f, 0.00001f );
+    BOOST_CHECK_THROW( synapses[3].getGID(), std::runtime_error );
+}
+
+BOOST_AUTO_TEST_CASE( efferent_synapses )
+{
+    const brain::Circuit circuit( brion::URI( BBP_TEST_BLUECONFIG3 ));
+    const brain::Synapses& synapses =
+            circuit.getEfferentSynapses( brion::GIDSet{10});
+    BOOST_CHECK( !synapses.empty( ));
+    BOOST_CHECK_EQUAL( synapses.size(), 74 );
+    BOOST_CHECK_EQUAL( synapses[0].getPostsynapticGID(), 1 );
+    BOOST_CHECK_CLOSE( synapses[1].getPostsynapticDistance(), 1.34995711f,
+                       0.00001f );
+    BOOST_CHECK_CLOSE( synapses[2].getConductance(), 0.34758395f, 0.00001f );
+    BOOST_CHECK_THROW( synapses[3].getGID(), std::runtime_error );
+}
+
+BOOST_AUTO_TEST_CASE( retrograde_projection )
+{
+    const brain::Circuit circuit( brion::URI( BBP_TEST_BLUECONFIG3 ));
+    const brion::GIDSet& preNeurons = circuit.getGIDs( "Layer1" );
+    const brion::GIDSet postNeuron = { 1 };
+    const brain::Synapses& synapses =
+            circuit.getProjectedSynapses( preNeurons, postNeuron );
+    BOOST_CHECK( !synapses.empty( ));
+    BOOST_CHECK_EQUAL( synapses.size(), 5 );
+    for( const auto& synapse : synapses )
+        BOOST_CHECK_EQUAL( synapse.getPresynapticGID(), 10 );
+}
+
+BOOST_AUTO_TEST_CASE( lazy_loading )
+{
+    const brain::Circuit circuit( brion::URI( BBP_TEST_BLUECONFIG3 ));
+    const brain::Synapses& synapses =
+            circuit.getAfferentSynapses( circuit.getGIDs( "Layer1" ),
+                                         brain::SYNAPSEPREFETCH_NONE );
+    BOOST_CHECK_EQUAL( synapses.size(), 1172 );
+    BOOST_CHECK_EQUAL( synapses[0].getPresynapticGID(), 10 );
+    BOOST_CHECK_CLOSE( synapses[1].getPostsynapticDistance(), 1.34995711f,
+                       0.00001f );
+    BOOST_CHECK_CLOSE( synapses[2].getConductance(), 0.34758395f, 0.00001f );
+    BOOST_CHECK_EQUAL( synapses[0].getPostsynapticCenterPosition(),
+                brion::Vector3f( 3.799289703f, 1947.041748047f, 9.237932205f ));
```

Comment:
Compare the data to another container loaded with prefetching, that makes the test more resilient to changes.